library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/mixin/client/ArmorFeatureRendererMixin.java

```diff
@@ -82,25 +79,87 @@ public abstract class ArmorFeatureRendererMixin {
+	@Inject(
+			method = "getArmorTexture",
+			at = @At(
+					value = "INVOKE",
+					target = "Ljava/util/Map;computeIfAbsent(Ljava/lang/Object;Ljava/util/function/Function;)Ljava/lang/Object;"
+			),
+			cancellable = true
+	)
+	private void quilt$getArmorTexture(ArmorItem item, boolean useSecondLayer, @Nullable String suffix, CallbackInfoReturnable<Identifier> cir) {
 		ItemStack stack = this.quilt$capturedEntity.getEquippedStack(this.quilt$capturedSlot);
 
 		Identifier texture = ARMOR_TEXTURE_CACHE.computeIfAbsent(
-				armorItem.getMaterial().getTexture()
-						+ ArmorTextureUtils.getArmorTextureSuffix(useSecondTexture, suffix)
+				item.getMaterial().getTexture()
+						+ ArmorTextureUtils.getArmorTextureSuffix(useSecondLayer, suffix)
 						+ ".png",
 				Identifier::new);
 		texture = ArmorRenderingRegistryImpl.getArmorTexture(texture, this.quilt$capturedEntity, stack, this.quilt$capturedSlot,
-				useSecondTexture, suffix);
+				useSecondLayer, suffix);
 
-		if (!texture.getPath().contains(".")) {
+		String textureString = texture.toString();
+		if (!textureString.contains(".")) {
 			LOGGER.warn("Armor texture identifier '" + texture + "' is missing file extension, automatically appending '.png'");
-			texture = new Identifier(texture.getNamespace(), texture.getPath() + ".png");
+			textureString += ".png";
 		}
 
-		cir.setReturnValue(ARMOR_TEXTURE_CACHE.computeIfAbsent(texture.toString(), Identifier::new));
+		// save armor texture for modifyArmorRenderLayer
+		cir.setReturnValue(this.quilt$capturedArmorTexture = ARMOR_TEXTURE_CACHE.computeIfAbsent(textureString, Identifier::new));
+	}
+
+	@ModifyArg(
+			method = "renderArmorParts",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/client/render/item/ItemRenderer;getArmorGlintConsumer(Lnet/minecraft/client/render/VertexConsumerProvider;Lnet/minecraft/client/render/RenderLayer;ZZ)Lcom/mojang/blaze3d/vertex/VertexConsumer;"
+			),
+			index = 1
+	)
+	private RenderLayer quilt$modifyArmorRenderLayer(RenderLayer layer) {
+		return ArmorRenderingRegistryImpl.getArmorRenderLayer(layer,
+				this.quilt$capturedEntity,
+				this.quilt$capturedEntity.getEquippedStack(this.quilt$capturedSlot),
+				this.quilt$capturedSlot,
+				this.quilt$capturedArmorTexture);
+	}
+
+	@ModifyArg(
+			method = "renderArmor(Lnet/minecraft/item/ArmorMaterial;Lnet/minecraft/client/util/math/MatrixStack;Lnet/minecraft/client/render/VertexConsumerProvider;ILnet/minecraft/item/trim/ArmorTrimPermutation;ZLnet/minecraft/client/render/entity/model/BipedEntityModel;ZFFF)V",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/client/texture/SpriteAtlasTexture;getSprite(Lnet/minecraft/util/Identifier;)Lnet/minecraft/client/texture/Sprite;"
+			),
+			index = 0
+	)
+	private Identifier quilt$modifyArmorTrimTexture(Identifier texture) {
+		// TODO
```

Comment:
hm, what is this TODO?

Reply:
Armor trim texture modification, possibly.